pyroute2/dhcp/hooks.py

```diff
@@ -24,21 +25,54 @@ async def unbound(self, lease: Lease):
 
 class ConfigureIP(Hook):
     async def bound(self, lease: Lease):
-        LOG.info('STUB: add %s to %s', lease.ip, lease.interface)
-        # await ip(
-        #     "addr",
-        #     "replace",
-        #     f"{lease.ip}/{lease.subnet_mask}",
-        #     "dev",
-        #     lease.interface,
-        # )
+        LOG.debug('Adding IP %s to %s', lease.ip, lease.interface)
+        async with AsyncIPRoute() as ipr:
+            await ipr.addr(
+                "replace",
+                index=await ipr.link_lookup(ifname=lease.interface),
+                address=lease.ip,
+                mask=lease.subnet_mask,
```

Comment:
Although we should definitely add some code to prevent the whole client from crashing if a hook fails, it would also be better to be more defensive (at least raising an exception with some context) for predictable failures like a missing option. Like for example a useful error message if the subnet mask is not provided.